### PR TITLE
Using router example improvements

### PR DESCRIPTION
**Pull request by timneutkens**
This example wasn't functioning :cry:

### CODE REVIEW

File: examples/using-router/components/Header.js [hunk 1]
```diff
@@ -2,30 +2,30 @@ import Router from 'next/router'
...
-  >
-    { children }
-  </a>
-)
 
-const styles = {
-  a: {
-    marginRight: 10
+function onClickHandler (href) {
+  return (e) => {
```
Comment (albinekb) [inline examples/using-router/components/Header.js]: I would call this `event`

File: examples/using-router/pages/error.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 
-const ErrorPage = ({ aa }) => (
-  <div>
-    <Header />
-    <p>This should not be rendered via SSR</p>
-  </div>
-)
+export default class extends Component {
+  static getInitialProps () {
+    console.log(Router.pathname)
```
Comment (albinekb) [inline examples/using-router/pages/error.js]: is this intentional?

File: examples/using-router/pages/error.js [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 import Router from 'next/router'
 
-const ErrorPage = ({ aa }) => (
-  <div>
-    <Header />
-    <p>This should not be rendered via SSR</p>
-  </div>
-)
+export default class extends Component {
+  static getInitialProps () {
```
Comment (albinekb) [inline examples/using-router/pages/error.js]: Maybe remove this completely?

File: examples/using-router/pages/error.js (hunk 2 above)
Comment (timneutkens) [inline examples/using-router/pages/error.js]: I believe it is intentionally broken to show what will happen when you use Router the wrong way.

File: examples/using-router/components/Header.js (hunk 1 above)
Comment (timneutkens) [inline examples/using-router/components/Header.js]: The other examples use `e` too, that's why I'm using it here 👍

Comment (albinekb) [inline examples/using-router/components/Header.js]: Very confusing..

File: examples/using-router/pages/error.js (hunk 2 above)
Comment (albinekb) [inline examples/using-router/pages/error.js]: hehe ok  😂

File: examples/using-router/components/Header.js (hunk 1 above)
Comment (timneutkens) [inline examples/using-router/components/Header.js]: True, think we could update them later on.

**Comment by arunoda** [comment]
Okay. This looks nice to me.

**Pull request merged**